src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt

```diff
@@ -0,0 +1,176 @@
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Group not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun getGroupDetail(
+    @Parameter(description = "The group code of the group.", required = true)
+    @PathVariable
+    group: String
+  ): GroupDetails {
+    return groupsService.getGroupDetail(group)
+  }
+
+  @PutMapping("/groups/child/{group}")
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_OAUTH_USERS')")
+  @Operation(
+    summary = "Amend child group name.",
+    description = "Amend a Child Group Name"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Child Group not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun amendChildGroupName(
+    @Parameter(description = "The group code of the child group.", required = true)
+    @PathVariable
+    group: String,
+    @Parameter(
+      description = "Details of the child group to be updated.",
+      required = true
+    ) @RequestBody
+    groupAmendment: GroupAmendment
+
+  ) {
+    groupsService.updateChildGroup(group, groupAmendment)
+  }
+}
+
+@Schema(description = "Group Name")
+data class GroupAmendment(
+  @Schema(required = true, description = "Group Name", example = "HDC NPS North East")
+  @field:NotBlank(message = "parent group code must be supplied")
+  val groupName: String
+)
+
+@Schema(description = "User Role")
+data class AuthUserAssignableRole(
+  @Schema(required = true, description = "Role Code", example = "LICENCE_RO")
+  val roleCode: String,
+
+  @Schema(required = true, description = "Role Name", example = "Licence Responsible Officer")
+  val roleName: String,
+
+  @Schema(required = true, description = "automatic", example = "TRUE")
+  val automatic: Boolean
+) {
+
+  constructor(a: Authority, automatic: Boolean) : this(a.roleCode, a.roleName, automatic)
+}
+
+@Schema(description = "User Role")
+data class Authority(
+  @Schema(required = true, description = "Role Code", example = "LICENCE_RO")
+  val roleCode: String,
+
```

Comment:
This class is the same as "Role" - we should probably just use one of them

Reply:
now removed this Authority not used in these endpoint...as suggested will use Role from RolesController